code/modules/integrated_electronics/core/helpers.dm

```diff
@@ -55,6 +58,10 @@
 	return
 
 /datum/integrated_io/proc/get_data()
+	if(islist(data))
+		for(var/i in 1 to length(data))
+			if(isweakref(data[i]))
+				data[i] = data[i].resolve()
```

Comment:
Did you tested it? There are very much resolve code across all the circuits. And if you will double call this proc,you'll get runtime.

Reply:
ok